WalletWasabi/WabiSabi/Client/Batching/PendingPayment.cs

```diff
@@ -0,0 +1,20 @@
+using System.Collections.Immutable;
+using NBitcoin;
+using WalletWasabi.Extensions;
+using WalletWasabi.WabiSabi.Models;
+
+namespace WalletWasabi.WabiSabi.Client.Batching;
+
+// Represents a payment that is waiting to be made. 
+public record PendingPayment(IDestination Destination, Money Amount)
+	: Payment(Destination, Amount)
+{
+	public Guid Id { get; } = Guid.NewGuid();
```

Comment:
It depends on how we want to design the system. In my mind all the payments (`Pending`, `InProgress` and `Finished`) represent the different states of a single payment and then, their ID is he exact same for all of them. However, if we simply replace `one state by the next one we will lose very important information because we will never know exactly what happened with each individual payment. 

IMO it is much better for each payment state to have a reference to its previous state so we can have the whole history of each payment. In that case, only the initial state should have the ID while all the rest would simply have a calculated property:

```c#
Public Guid Id => previousState.Id;
```

-----------

This is not the only property missing, each payment state should also have a `StartTime` property, 'Reason` in case of those that failed and maybe others.